internal/builds/builds_test.go

```diff
@@ -96,13 +92,35 @@ func (suite *BuildsIntegrationTestSuite) TestGetByID() {
 	})
 }
 
+func (suite *BuildsIntegrationTestSuite) TestGetByVersionString() {
+	suite.Run("successful looks up existing build by version string", func() {
+
+		// create a build instance to look up
+		existingBuild, err := suite.app.builds.CreateNew(suite.goodCreateBuildRequest)
+		suite.Require().NoError(err)
+
+		result, err := suite.app.builds.GetByVersionString(suite.goodCreateBuildRequest.VersionString)
+		suite.Assert().NoError(err)
+
+		// make sure the id's match
+		suite.Assert().Equal(existingBuild.ID, result.ID)
+	})
+
+	suite.Run("errors not found for non-existent version string", func() {
+
+		_, err := suite.app.builds.GetByVersionString("does-not-exist")
+		suite.Assert().ErrorIs(err, ErrBuildNotFound)
+	})
+}
+
 type testApplication struct {
 	builds *BuildController
 	db     *gorm.DB
 }
 
 func initTestApp(t *testing.T) *testApplication {
 	dbConn := testutils.ConnectAndMigrate(t)
+	dbConn = dbConn.Begin(&sql.TxOptions{Isolation: sql.LevelSerializable})
```

Comment:
I never ended up pushing sql-shenanigans in my last commit, but I was reading around and I think pgx implements anything database/sql and since that's our driver anyway we should prefer that. 

Also, is this movable to testutils like ConnectAndMigrate? It's a little bit scary-sql-thing that probably would scare people.